imap_processing/cdf/tests/test_cdf_attribute_manager.py

```diff
@@ -1,58 +1,322 @@
+        assert attr in cdf_manager.global_attribute_schema.keys()
 
-    mag_l1a_global_attrs = cdf_manager.get_global_attributes("imap_mag_l1a_norm-raw")
 
+def test_get_global_attributes():
+    """
+    Test function that covers:
+        get_global_attributes
+    """
+    # Initialize CdfAttributeManager object which loads in default info
+    cdf_manager = CdfAttributeManager(Path(__file__).parent.parent / "config")
+
+    # Change filepath to load test global attributes
+    cdf_manager.source_dir = Path(__file__).parent.parent / "tests"
+    cdf_manager.load_global_attributes("imap_default_global_test_cdf_attrs.yaml")
+    cdf_manager.load_global_attributes("imap_test_global.yaml")
+
+    # Loading in instrument specific attributes
+    test_get_global_attrs = cdf_manager.get_global_attributes("imap_test_T1_test")
+
+    # Testing information previously loaded into global attributes (first if statement)
+    assert test_get_global_attrs["Project"] == "STP>Solar-Terrestrial Physics"
     assert (
-        mag_l1a_global_attrs["Mission_group"]
+        test_get_global_attrs["Source_name"]
         == "IMAP>Interstellar Mapping and Acceleration Probe"
     )
-    assert mag_l1a_global_attrs["Descriptor"] == "MAG>Magnetometer"
-    assert mag_l1a_global_attrs["Logical_source"] == "imap_mag_l1a_norm-raw"
+    assert test_get_global_attrs["Mission_group"] == "Dysfunctional Cats"
+    # Testing instrument specific global attributes (first elif statement)
+    assert test_get_global_attrs["Descriptor"] == "TEST>Testinstrument"
+    assert test_get_global_attrs["Data_type"] == "T1_test-one>Test-1 test one"
+    assert test_get_global_attrs["Logical_source"] == "imap_test_T1_test"
+    assert (
+        test_get_global_attrs["Logical_source_description"]
+        == "IMAP Mission TEST one document Level-T1."
+    )
+    # Not given, and not required information
+    assert test_get_global_attrs["Data_level"] is None
+    with pytest.raises(KeyError):
+        assert test_get_global_attrs["bad_name"] == "false info"
+
+    # Testing second elif statement
+    test_error_elif = cdf_manager.get_global_attributes("imap_test_T3_test")
+    with pytest.raises(KeyError):
+        assert test_error_elif["Data_type"] == "Does Not Exist"
+
+    # Load in more data using get_global_attributes
+    test_get_global_attrs_2 = cdf_manager.get_global_attributes("imap_test_T2_test")
+    # Testing information previously loaded into global attributes (first if statement)
+    assert test_get_global_attrs_2["Project"] == "STP>Solar-Terrestrial Physics"
+    # Testing first elif statement
+    assert test_get_global_attrs_2["Descriptor"] == "TEST>Testinstrument"
+    # "Data_type" not required according to default schema
+    assert test_get_global_attrs_2["Data_type"] == "T2_test-two>Test-2 test two"
+
+    # Testing how instrument_id operates
+    assert test_get_global_attrs["Project"] == cdf_manager.global_attributes["Project"]
```

Comment:
are these meant to be `test_get_global_attrs_2`? If not, nitpick, but move them into the section with `test_get_global_attrs`